internal/tunnel/proto/msg.go

```diff
@@ -72,8 +72,17 @@ type AuthExtra struct {
 	ConfigVersion   string
 	CustomInterface bool
 	CustomCAs       bool
+
+	ClientType ClientType // The type of client this is. Currently agent and library clients are supported
```

Comment:
instead of Agent | Library, should this be Agent | GolangLibrary | JavaLibrary | etc?

Reply:
Mmmm, maybe?

Realistically, I think we'll have a different protocol altogether before we get may more libraries, so just agent/library is probably fine for right now.

Reply:
i think it's going to be better long term to just treat this as a "User Agent" where each language is different so that it can be unique across each of them. it's not any extra handling server side and it gives us more fidelity

Reply:
OK, I'm packing more info into it over in ngrok-private/ngrok#13848. Any thoughts on the information/structure there?

Reply:
The `official` thing seems kinda.. I dunno, redundant?
Would it ever _not_ be marked official?

Reply:
Not by us, but since the library is open-source, nothing stops people from forking it and creating unofficial ngrok libraries.

We also can't stop them from simply continuing to claim that they're official, but my thinking is that this at least gives them the opportunity to play nice.